staging/src/k8s.io/client-go/tools/cache/store.go

```diff
@@ -109,6 +109,11 @@ func MetaNamespaceKeyFunc(obj interface{}) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("object has no meta: %v", err)
 	}
+	// When statefulset pods update, the old pod shares the same name with new one
+	// We should use UID to distinguish them.
+	if len(meta.GetUID()) > 0 {
+		return string(meta.GetUID()), nil
+	}
```

Comment:
There is another `SplitMetaNamespaceKey` below, which is the reverse of this. If UID is used, that `SplitMetaNamespaceKey` will be invalid

Reply:
All the 213 unit test failures are related to this change...

@deads2k @liggitt - you might have thoughts about this